src/js/core/modal_dialog_elements.js

```diff
@@ -366,20 +366,27 @@ export class DialogWithForm extends Dialog {
      * @param {string} param0.title
      * @param {string} param0.desc
      * @param {string=} param0.confirmButton
+     * @param {string=} param0.extraButton
      * @param {Array<FormElement>} param0.formElements
      */
-    constructor({ app, title, desc, formElements, confirmButton = "ok:good" }) {
+    constructor({ app, title, desc, formElements, confirmButton = "ok:good", extraButton = null }) {
```

Comment:
I don't like extraButton - Why not just make the buttons itself a parameter? E.g. `buttons = ["cancel", "ok:good"]` that way we could also get rid of `confirmButton`

Reply:
https://github.com/tobspr/shapez.io/blob/2722366d686fd551a8edceebafc8e78c1e40e976/src/js/core/modal_dialog_elements.js#L373
I agree